pkg/api/pod/util.go

```diff
@@ -495,7 +495,7 @@ func dropDisabledFields(
 		}
 	}
 
-	if !utilfeature.DefaultFeatureGate.Enabled(features.ProbeTerminationGracePeriod) && !probeGracePeriodInUse(oldPodSpec) {
+	if !probeGracePeriodInUse(oldPodSpec) {
```

Comment:
I think this is the problem.

We should never drop terminationGracePeriodSeconds of a probe.

I'll propose a fix for this.

Reply:
https://github.com/kubernetes/kubernetes/pull/119354

Reply:
you are correct... this block should have gone away entirely